django-nonrel/ocl/concepts/validators.py

```diff
@@ -127,3 +133,8 @@ def description_cannot_be_null(self, concept):
 
         if len(empty_descriptions):
             raise ValidationError({'descriptions': [BASIC_DESCRIPTION_CANNOT_BE_EMPTY]})
+
+    def custom_attribute_key_must_be_encoded(self, concept):
+        concept.encode_extras()
```

Comment:
It's not correct to encode inside a validator. Validators are read-only and should not modify model.